Microsoft.Toolkit.Uwp.UI.Controls/Menu/MenuItem.cs

```diff
@@ -0,0 +1,270 @@
+
+        /// <summary>
+        /// Identifies the <see cref="Header"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty HeaderProperty = DependencyProperty.Register(nameof(Header), typeof(string), typeof(MenuItem), new PropertyMetadata(default(string)));
+
+        /// <summary>
+        /// Gets or sets the title to appear in the title bar
+        /// </summary>
+        public string Header
+        {
+            get { return (string)GetValue(HeaderProperty); }
+            set { SetValue(HeaderProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the menu is opened or not
+        /// </summary>
+        public bool IsOpened
+        {
+            get
+            {
+                return _isOpened;
+            }
+
+            private set
+            {
+                _parentMenu.IsOpened = _isOpened = value;
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MenuItem"/> class.
+        /// </summary>
+        public MenuItem()
+        {
+            DefaultStyleKey = typeof(MenuItem);
+            IsFocusEngagementEnabled = true;
+        }
+
+        /// <summary>
+        /// This method is used to show the menu for current item
+        /// </summary>
+        public void ShowMenu()
+        {
+            FlyoutButton?.Flyout?.ShowAt(FlyoutButton);
+        }
+
+        /// <summary>
+        /// This method is used to hide the menu for current item
+        /// </summary>
+        public void HideMenu()
+        {
+            FlyoutButton?.Flyout?.Hide();
+        }
+
+        /// <inheritdoc />
+        protected override void OnApplyTemplate()
+        {
+            FlyoutButton = GetTemplateChild(FlyoutButtonName) as Button;
+            _parentMenu = this.FindAscendant<Menu>();
+            IsOpened = false;
+
+            Items.VectorChanged -= Items_VectorChanged;
+
+            if (_menuFlyout == null)
+            {
+                _menuFlyout = new MenuFlyout();
+            }
+            else
+            {
+                _menuFlyout.Opened -= MenuFlyout_Opened;
+                _menuFlyout.Closed -= MenuFlyout_Closed;
+            }
+
+            if (FlyoutButton != null)
+            {
+                FlyoutButton.PointerExited -= FlyoutButton_PointerExited;
+                Items.VectorChanged += Items_VectorChanged;
+
+                _menuFlyout.Placement = _parentMenu.Orientation == Orientation.Horizontal
+                    ? FlyoutPlacementMode.Bottom
+                    : FlyoutPlacementMode.Right;
+
+                _menuFlyout.Opened += MenuFlyout_Opened;
+                _menuFlyout.Closed += MenuFlyout_Closed;
+                FlyoutButton.PointerExited += FlyoutButton_PointerExited;
+
+                _menuFlyout.MenuFlyoutPresenterStyle = _parentMenu.MenuFlyoutStyle;
+                FlyoutButton.Style = _parentMenu.HeaderButtonStyle;
```

Comment:
Do we need the HeaderButtonStyle? Can developers just style the MenuItem itself? Seems an odd way to style a control